tests/integration_tests/functional/test_cpu_features_x86_64.py

```diff
@@ -1149,3 +1149,23 @@ def test_intel_amx_reported_on_sapphire_rapids(
         "=",
         expected_dict,
     )
+
+
+def test_waitpkg_inaccessibility(uvm_nano, waitpkg_bin):
+    """
+    Verifies that attempting to use WAITPKG (UMONITOR / UMWAIT instructions)
+    generates #UD.
+    """
+    vm = uvm_nano
+    vm.add_net_iface()
+    vm.start()
+
+    rmt_path = "/tmp/waitpkg"
+    vm.ssh.scp_put(waitpkg_bin, rmt_path)
+
+    ret, _, _ = vm.ssh.run(rmt_path)
+    assert ret == 255
```

Comment:
mh, 255 seems weird, as that's the code that SSH returns if something went wrong with the ssh connection itself :thinking:

Reply:
I managed to fix it by printing the return value :)
I don't know why ssh command crashes if illegal instruction happens though.

Reply:
it returns 255...
```
['ip', 'netns', 'exec', 'netns-master-0', 'ssh', '-o', 'ControlPath=/srv/jailer/firecracker/d4481d01-216f-41ce-8f94-eab4d9a2d51d/root/ssh-0.sock', 'root@192.168.0.2', 'bash -c /tmp/waitpkg']
[294] stdout:

[294] stderr:

Returned error code: 255
```